airflow/jobs.py

```diff
@@ -1891,118 +1925,92 @@ def _manage_executor_state(self, started):
                     self.logger.error(msg)
                     ti.handle_failure(msg)
 
-    def _execute(self):
+    @provide_session
+    def _tasks_instances_for_run_date(self, run_date, session=None):
```

Comment:
I thought it'd be better to actually split this function into two functions, preserving the way queries are executed, avoiding tuple return values.